core/esmf-aspect-meta-model-java/src/main/java/org/eclipse/esmf/aspectmodel/loader/Instantiator.java

```diff
@@ -162,6 +163,37 @@ protected Value buildValue( final RDFNode node, final Optional<Resource> charact
                .orElseThrow( () -> new AspectLoadingException( "Literal can not be parsed: " + literal ) );
       }
 
+      if ( node.isResource() ) {
+         Resource resource = node.asResource();
+
+         if ( resource.canAs( RDFList.class ) ) {
+            RDFList list = resource.as( RDFList.class );
+            List<Value> values = list.iterator().toList().stream()
+                  .map( rdfNode -> buildValue( rdfNode, characteristicResource, type ) )
+                  .toList();
+
+            return new DefaultCollectionValue( values, CollectionValue.CollectionType.LIST, type );
+         }
```

Comment:
I think this check needs to be removed. An RDF list does not always imply CollectionType.LIST. Consider the following example:

```turtle
:E a samm:Entity ;
  samm:properties ( :eprop ) .

:eprop a samm:Property ;
  samm:characteristic [
    a samm-c:Set ;
    samm:dataType xsd:integer ;
  ] .

:MyE a :E ;
  :eprop ( 1 2 3 ) .
```

In this case, the value for eprop in MyE must have CollectionType.SET, not CollectionType.LIST. The check based on the Characteristic takes place further down below.